### PR TITLE
build: Update Bouncycastle to 1.77.0

### PR DESCRIPTION
**Pull request by MMaiero**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** #5025

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped

**Screenshots:** If applicable, add screenshots to help explain your solution

**Manual Tests**: Optional description of the tests performed to check correct functioning of changes, useful for an efficient review

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

File: NOTICE.md [hunk 1]
```diff
@@ -124,11 +124,11 @@ This project leverages the following third party content.
 * maven/mavencentral/org.apache.servicemix.bundles/org.apache.servicemix.bundles.spring-expression/4.3.20.RELEASE_1, Apache-2.0, approved, CQ16242
 * maven/mavencentral/org.apache.servicemix.bundles/org.apache.servicemix.bundles.spring-jms/4.3.20.RELEASE_1, Apache-2.0, approved, CQ16243
 * maven/mavencentral/org.apache.servicemix.bundles/org.apache.servicemix.bundles.spring-tx/4.3.20.RELEASE_1, Apache-2.0, approved, CQ16244
-* maven/mavencentral/org.bouncycastle/bcpg-jdk18on/1.71, Apache-2.0, approved, #2665
```
Comment (nicolatimeus) [inline NOTICE.md:127]: The referenced iplab issues probably need update

**Comment by MMaiero** [comment]
Opened CQ: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11700

**Comment by salvatore-coppola** [comment]
@MMaiero why do we choose 1.74 and not 1.77 released two weeks ago ?

**Comment by MMaiero** [comment]
It is the minimal requirement to solve the reported CVEs. And more likely to already been approved

**Comment by MMaiero** [comment]
@salvatore-coppola @nicolatimeus Ok to merge?

**Comment by salvatore-coppola** [comment]
> @salvatore-coppola @nicolatimeus Ok to merge?

fine with me

File: kura/org.eclipse.kura.ai.triton.server/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -25,18 +25,18 @@ Import-Package: com.google.common.base;version="25.0.0",
...
+ org.bouncycastle.jcajce;version="1.77.0",
+ org.bouncycastle.jcajce.interfaces;version="1.77.0",
+ org.bouncycastle.jcajce.io;version="1.77.0",
+ org.bouncycastle.jcajce.spec;version="1.77.0",
+ org.bouncycastle.jcajce.util;version="1.77.0",
+ org.bouncycastle.jce.provider;version="1.77.0",
  org.bouncycastle.openpgp;version="1.71.0",
  org.bouncycastle.openpgp.jcajce;version="1.71.0",
  org.bouncycastle.openpgp.operator.jcajce;version="1.71.0",
- org.bouncycastle.util;version="1.68.0",
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.ai.triton.server/META-INF/MANIFEST.MF:37]: Do we want to update also the following versions for consistency?

```
 org.bouncycastle.openpgp;version="1.71.0",
 org.bouncycastle.openpgp.jcajce;version="1.71.0",
 org.bouncycastle.openpgp.operator.jcajce;version="1.71.0",
```

Comment (MMaiero) [reply kura/org.eclipse.kura.ai.triton.server/META-INF/MANIFEST.MF:37]: Oh! I have missed that!
I'll do it now

Comment (MMaiero) [reply kura/org.eclipse.kura.ai.triton.server/META-INF/MANIFEST.MF:37]: Done

**Review by nicolatimeus — APPROVED** [review]

**Pull request merged**